hub/api/tests/test_api.py

```diff
@@ -691,3 +694,31 @@ def test_htypes_list():
         "binary_mask",
         "segment_mask",
     ]
+
+
+def test_hierarchical_tensors(local_ds_generator):
+    ds = local_ds_generator()
+    ds.create_tensor("x")
+    with pytest.raises(TensorAlreadyExistsError):
+        ds.create_tensor("x/y")
+    ds.create_tensor("y/x")
+
```

Comment:
Following a convention for naming groups vs tensors might make code easier to read -- for example, capital letters for groups and small letters for tensors.